engines/cryomni3d/versailles/logic.cpp

```diff
@@ -0,0 +1,4650 @@
+	// Dispatch to the correct state
+	if (_gameVariables[GameVariables::kCollectFood]) {
+		// Draw plate with less food
+		ZonFixedImage::CallbackFunctor *functor =
+		    new Common::Functor1Mem<ZonFixedImage *, void, CryOmni3DEngine_Versailles>(this,
+		            &CryOmni3DEngine_Versailles::img_44071b);
+		fimg->changeCallback(functor);
+		return;
+	}
+
+	// There we have a full plate
+	fimg->load("41B_bboy.GIF");
+	while (1) {
+		fimg->manage();
+		if (fimg->_exit || fimg->_zoneLow) {
+			fimg->_exit = true;
+			break;
+		}
+		if (fimg->_zoneUse) {
+			// Collected food
+			collectObject(124, fimg);
+			_gameVariables[GameVariables::kCollectFood] = 1;
+			// Draw plate with less food
+			ZonFixedImage::CallbackFunctor *functor =
+			    new Common::Functor1Mem<ZonFixedImage *, void, CryOmni3DEngine_Versailles>(this,
+			            &CryOmni3DEngine_Versailles::img_44071b);
+			fimg->changeCallback(functor);
+			break;
+		}
+	}
+}
+
+IMG_CB(44071b) {
+	// There we have less food on plate
+	fimg->load("41B_bbo2.GIF");
+	while (1) {
+		fimg->manage();
+		if (fimg->_exit || fimg->_zoneLow) {
+			fimg->_exit = true;
+			break;
+		}
+	}
+}
+
+IMG_CB(44161) {
+	// Dispatch to the correct state
+	if (_gameVariables[GameVariables::kCollectQuill] == 1 && !_inventory.inInventoryByNameID(126)) {
+		// We have collected quill but not solved epigraph yet
+		ZonFixedImage::CallbackFunctor *functor =
+		    new Common::Functor1Mem<ZonFixedImage *, void, CryOmni3DEngine_Versailles>(this,
+		            &CryOmni3DEngine_Versailles::img_44161b);
+		fimg->changeCallback(functor);
+		return;
+	} else if (_gameVariables[GameVariables::kUsedVaubanBlueprint1] == 1 &&
+	           _gameVariables[GameVariables::kUsedVaubanBlueprint2] == 1) {
+		// We have used vauban blueprints: display the solution
+		ZonFixedImage::CallbackFunctor *functor =
+		    new Common::Functor1Mem<ZonFixedImage *, void, CryOmni3DEngine_Versailles>(this,
+		            &CryOmni3DEngine_Versailles::img_44161f);
+		fimg->changeCallback(functor);
+		return;
+	} else if (_gameVariables[GameVariables::kCollectQuill] == 1 &&
+	           _inventory.inInventoryByNameID(126)) {
+		// We have collected quill and epigraph
+		ZonFixedImage::CallbackFunctor *functor =
+		    new Common::Functor1Mem<ZonFixedImage *, void, CryOmni3DEngine_Versailles>(this,
+		            &CryOmni3DEngine_Versailles::img_44161c);
+		fimg->changeCallback(functor);
+		return;
+	}
+
+	// There we have blueprints, quill and epigraph on desk
+	fimg->load("42X2_20.GIF");
+	while (1) {
+		fimg->manage();
+		if (fimg->_exit || fimg->_zoneLow) {
+			fimg->_exit = true;
+			break;
+		}
+		if (fimg->_zoneUse && fimg->_currentZone == 0 && currentGameTime() >= 2) {
+			// Collected quill
+			collectObject(128, fimg);
+			_gameVariables[GameVariables::kCollectQuill] = 1;
+			// Try to solve epigraph
+			ZonFixedImage::CallbackFunctor *functor =
+			    new Common::Functor1Mem<ZonFixedImage *, void, CryOmni3DEngine_Versai
```

Comment:
consider using `FROM_BE_32(_countdownValue)`

Reply:
I assume this review is for when I mangle _countdownValue and the game variable.
It's not exactly a swap between bytes but more a conversion from an int to a char array with a colon in the middle so using FROM_BE_32 would not do what is intended.

Reply:
Okay.